src/libaktualizr/package_manager/dockerappmanager_test.cc

```diff
@@ -0,0 +1,122 @@
+                                                    repo_path, port);
+  TestUtils::waitForServer(repo_server + "/");
+  return p;
+}
+
+TEST(DockerAppManager, PackageManager_Factory_Good) {
+  Config config;
+  config.pacman.type = PackageManager::kOstreeDockerApp;
+  config.pacman.sysroot = test_sysroot;
+  TemporaryDirectory dir;
+  config.storage.path = dir.Path();
+
+  std::shared_ptr<INvStorage> storage = INvStorage::newStorage(config.storage);
+  auto pacman = PackageManagerFactory::makePackageManager(config.pacman, storage, nullptr, nullptr);
+  EXPECT_TRUE(pacman);
+}
+
+TEST(DockerAppManager, DockerApp_Fetch) {
+  std::string sha = Utils::readFile(test_sysroot / "ostree/repo/refs/heads/ostree/1/1/0", true);
+  Json::Value target_json;
+  target_json["hashes"]["sha256"] = sha;
+  target_json["custom"]["targetFormat"] = "OSTREE";
+  target_json["length"] = 0;
+  target_json["custom"]["docker_apps"]["app1"]["filename"] = "primary.txt";
+  Uptane::Target target("pull", target_json);
+
+  TemporaryDirectory temp_dir;
+  auto repo = temp_dir.Path();
+  auto repod = create_repo(repo);
+
+  Config config;
+  config.pacman.type = PackageManager::kOstreeDockerApp;
+  config.pacman.sysroot = test_sysroot;
+  config.pacman.docker_apps_root = temp_dir / "docker_apps";
+  config.pacman.docker_apps.push_back("app1");
+  config.pacman.docker_app_bin = config.pacman.docker_compose_bin = "src/libaktualizr/package_manager/docker_fake.sh";
+  config.pacman.ostree_server = treehub_server;
+  config.uptane.repo_server = repo_server + "/repo/image";
+  TemporaryDirectory dir;
+  config.storage.path = dir.Path();
+
+  std::shared_ptr<INvStorage> storage = INvStorage::newStorage(config.storage);
+  KeyManager keys(storage, config.keymanagerConfig());
+  auto http = std::make_shared<HttpClient>();
+  auto client = SotaUptaneClient::newTestClient(config, storage, http, nullptr);
+  ASSERT_TRUE(client->updateImagesMeta());
+
+  std::string targets = Utils::readFile(repo / "repo/image/targets.json");
+  LOG_INFO << "Repo targets " << targets;
+
+  bool result = client->package_manager_->fetchTarget(target, *(client->uptane_fetcher), keys, progress_cb, nullptr);
+  ASSERT_TRUE(result);
+
+  auto hashes = std::vector<Uptane::Hash>{
+      Uptane::Hash(Uptane::Hash::Type::kSha256, "A06AC4D8F2C389DC0F919B6BA2A809324C0D3E368741EC210BE34DB8179EEBB7"),
+      Uptane::Hash(Uptane::Hash::Type::kSha512,
+                   "C2F598F7CDD0C4EA8D23A277000E1CE413C2FCAA456E980134611854BF618A317F8280CF517E2DF5BAD73FC11666884CD8B"
+                   "5CADFA7F9D2C4B4B1B84E3574C4CE")};
+  Uptane::Target app_target("primary.txt", hashes, 8);
+  ASSERT_TRUE(storage->checkTargetFile(app_target));
+
+  client->package_manager_->install(target);
+  std::string content = Utils::readFile(config.pacman.docker_apps_root / "app1/docker-compose.yml");
+  ASSERT_EQ("DOCKER-APP RENDER OUTPUT\nprimary\n", content);
```

Comment:
I think the fact that the docker-app file being used is named "primary.txt" is confusing me. So is this actually testing that you can install a file with that name as a docker app? I think an explanation of what this test is specifically covering would be really helpful. Historically, we've linked test comments to items in the actions.md file to track what is covered by tests. That might get revised, but either way, at least having a comment preceding this test to explain what makes it interesting would be great.

Reply:
I'd just copy/pasted that from another test in this repository. I can fix that.